classes/submissionFile/Collector.inc.php

```diff
@@ -0,0 +1,253 @@
+    }
+
+    /**
+     * Set fileIds filter
+     */
+    public function filterByFileIds(?array $fileIds): self
+    {
+        $this->fileIds = $fileIds;
+
+        return $this;
+    }
+
+    /**
+     * Set assocType and assocId filters
+     *
+     * @param null|array $assocTypes One or more of the ASSOC_TYPE_ constants
+     * @param null|array $assocIds Match with ids for these assoc types
+     */
+    public function filterByAssoc(?array $assocTypes, ?array $assocIds = []): self
+    {
+        $this->assocTypes = $assocTypes;
+
+        if ($assocIds !== []) {
+            $this->assocIds = $assocIds;
+        }
+
+        return $this;
+    }
+
+    /**
+     * Set uploaderUserIds filter
+     */
+    public function filterByUploaderUserIds(?array $uploaderUserIds): self
+    {
+        $this->uploaderUserIds = $uploaderUserIds;
+
+        return $this;
+    }
+
+    /**
+     * Whether or not to include dependent files in the results
+     */
+    public function filterByIncludeDependentFiles(bool $includeDependentFiles): self
+    {
+        $this->includeDependentFiles = $includeDependentFiles;
+
+        return $this;
+    }
+
+    /**
+     * Limit the number of objects retrieved
+     */
+    public function limit(int $count): self
+    {
+        $this->count = $count;
+
+        return $this;
+    }
+
+    /**
+     * Offset the number of objects retrieved, for example to
+     * retrieve the second page of contents
+     */
+    public function offset(int $offset): self
+    {
+        $this->offset = $offset;
+
+        return $this;
+    }
+
+    /**
+     * @copydoc CollectorInterface::getQueryBuilder()
+     */
+    public function getQueryBuilder(): Builder
+    {
+        $qb = DB::table($this->dao->table . ' as sf')
+            ->select('sf.*');
+
+        if ($this->submissionIds !== null) {
+            $qb->whereIn('sf.submission_id', $this->submissionIds);
+        }
+
+        if ($this->fileStages !== null) {
+            $qb->whereIn('sf.file_stage', $this->fileStages);
+        }
+
+        if ($this->genreIds !== null) {
+            $qb->whereIn('sf.genre_id', $this->genreIds);
+        }
+
+        if ($this->fileIds !== null) {
+            $qb->join('submission_file_revisions as sfr', 'sfr.submission_file_id', '=', 'sf.submission_file_id')
+                ->whereIn('sfr.file_id', $this->fileIds);
+        }
+
+        if ($this->reviewRoundIds !== null) {
+            $qb->whereIn('sf.submission_file_id', function ($query) {
+                return $query->select('submission_file_id')
+                    ->from('review_round_files')
+                    ->whereIn('review_round_id', $this->reviewRoundIds);
+            });
+        }
+
+        if ($this->reviewIds !== null) {
+            $qb->join('review_files as rf', 'rf.submission_file_id', '=', 'sf.submission_file_id')
+                ->whereIn('rf.review_id', $this->reviewIds);
+        }
+
+        if ($this->assocTypes !== null) {
```

Comment:
This still needs fixing: https://github.com/pkp/pkp-lib/pull/7166#discussion_r676841825